lint/type-guard_checker.go

```diff
@@ -18,6 +23,67 @@ func NewTypeGuardChecker(ctx *Context) *TypeGuardChecker {
 
 // Check runs type switch checks for f.
 func (c *TypeGuardChecker) Check(f *ast.File) []Warning {
-	// TODO: implement me. Refs #10.
-	return nil
+	c.warnings = c.warnings[:0]
+	// TODO(quasilyte): do recursive analysis. Type switches
+	// may have other type switches inside their clauses that
+	// can trigger warnings as well.
+	for _, decl := range collectFuncDecls(f) {
+		ast.Inspect(decl.Body, func(x ast.Node) bool {
+			if stmt, ok := x.(*ast.TypeSwitchStmt); ok {
+				c.check(stmt)
+				return false
+			}
+			return x != nil
+		})
+	}
+	return c.warnings
```

Comment:
Warnings seems to never filled

Reply:
See `func (c *TypeGuardChecker) warn(stmt *ast.TypeSwitchStmt, caseIndex int) {`.
I'm following convention established in `param-name` checker.
`warn` prefixed methods append specific warning.
If kind is not empty, it's appended to method name, like `warnLoud` in case `Kind="Loud"`.